minirys:
  # Virtual Receptors configs
  battery_vr:
    ros__parameters:
      updateFrequency: 10.0
      cellMultiplier1: 2.55
      cellMultiplier2: 5.11
      cellMultiplier3: 7.68
      movingAverageWeight: 0.5
      undervoltageThresholdWarning: 3.3
      undervoltageThresholdError: 3.1

  temperature_vr:
    ros__parameters:
      updateFrequency: 2.0
      cpuTemperaturePath: "/sys/class/thermal/thermal_zone0/temp"

  imu_vr:
    ros__parameters:
      updateFrequency: 100.0
      filterFactor: 0.05
      angleCorrection: 0.000346 #(Yellow 0.000346) #-0.189545 orange 0.042729
      angleHistorySize: 4
      gyroOffsetX: 1.977383 #(Yellow 3.850143) #in deg
      gyroOffsetY: -6.768773 #(Yellow -5.236362) #in deg
      gyroOffsetZ: -2.870283 #(Yellow -4.604892) #in deg

  distance_vr:
    ros__parameters:
      updateFrequency: 10.0

  camera_vr:
    ros__parameters:
      image_size: [1920, 1080]
      white_balance_automatic: true

  # Virtual Effectors configs
  fan_ve:
    ros__parameters:
      updateFrequency: 2.0
      pwmFrequency: 10.0

  motors_ve:
    ros__parameters:
      updateFrequency: 100.0
      stepsPerRevolution: 200.0
      maxSpeed: 12.56 # 2*Pi = 1rps
      acceleration: 5000.0 # 2*Pi = 1rps

  # Control Subsystem
  fan_regulator_cs:
    ros__parameters:
      updateFrequency: 2.0
      fanLevelLow: 0.4
      fanLevelMedium: 0.8
      fanLevelHigh: 1.0
      thresholdCPUWarning: 50.0
      thresholdCPUCritical: 65.0
      thresholdMainWarning: 45.0
      thresholdMainCritical: 60.0

  motors_controller_cs:
    ros__parameters:
      updateFrequency: 100.0
      invertLeftMotor: true
      invertRightMotor: false
      enableSpeedRegulator: false
      maxLinearSpeed: 10.5
      maxRotationSpeed: 10.5
      maxWheelSpeed: 10.5
      maxStandUpSpeed: 6.0
      maxBalancingAngle: 0.25
      pidSpeedKp: 0.0
      pidSpeedKi: 0.0
      pidSpeedKd: 0.0
      pidAngleKp: 100.0   #100.0
      pidAngleKi: 1.61   #1.61
      pidAngleKd: 0.00001     #0.00001

  odometry_cs:
    ros__parameters:
      updateFrequency: 10.0
      invertLeftMotor: true
      invertRightMotor: false
      wheelRadius: 0.055
      wheelSeparation: 0.139
      wheelRadiusCorrection: 1.00
      wheelSeparationCorrection: 1.00

  minirys_rttest:
    ros__parameters:
      setMemoryStrategy: false
      lockMemory: false
      schedulerPriority: 80
      schedulerPolicy: "other"
      iterations: 120000 # = 20 minutes, 20[min] * 60[s/min] * 100[it/s]
      updatePeriod: 10000000 # [ns] = 10ms
      stackSizeMB: 10
      prefaultDynamicSizeMB: 512
      filename: "/tmp/minirys_rttest.results.txt"
